frontend/src/features/admin-form/create/logic/components/LogicContent/InactiveLogicBlock/FieldLogicBadge.tsx

```diff
@@ -9,30 +10,74 @@ import { FormFieldWithQuestionNo } from '~features/form/types'
 import { LogicBadge } from './LogicBadge'
 
 interface FieldLogicBadgeProps {
-  field: FormFieldWithQuestionNo
+  field?: FormFieldWithQuestionNo
+  defaults?: {
+    variant: 'error' | 'info'
+    message: string
+  }
 }
 
 /**
  * Field specific logic badge. Adds field icon and question number to the displayed badge.
  */
-export const FieldLogicBadge = ({ field }: FieldLogicBadgeProps) => {
+export const FieldLogicBadge = ({
+  field,
+  defaults = { variant: 'error', message: 'Field not found' },
+}: FieldLogicBadgeProps) => {
   const fieldMeta = useMemo(
-    () => BASICFIELD_TO_DRAWER_META[field.fieldType],
-    [field.fieldType],
+    () => (field ? BASICFIELD_TO_DRAWER_META[field.fieldType] : null),
+    [field],
   )
 
+  const tooltipLabel = useMemo(
+    () => (!fieldMeta ? defaults.message : `${fieldMeta.label} field`),
+    [fieldMeta, defaults.message],
+  )
+
+  const errorIcon = useMemo(() => {
+    switch (defaults.variant) {
+      case 'error':
+        return <Icon as={BxsErrorCircle} fontSize="1rem" color="danger.500" />
+      case 'info':
+        return <Icon as={BxsInfoCircle} fontSize="1rem" color="primary.500" />
+    }
+  }, [defaults.variant])
+
+  const errorText = useMemo(() => {
+    switch (defaults.variant) {
+      case 'error':
+        return (
+          <Text textColor="danger.500" noOfLines={1}>
+            {defaults.message}
+          </Text>
+        )
+      case 'info':
+        return (
+          <Text textColor="primary.500" noOfLines={1}>
+            {defaults.message}
+          </Text>
+        )
+    }
+  }, [defaults])
+
```

Comment:
Rather than duplicating the markup in these 2 blocks, how about storing the color as a local variable since it's the only thing that changes, and then place the jsx markup below?

Or is the current structure done on purpose with the possibility that 2 `useMemos()` get `undefined` in case of unexpected variants? Is that even possible?

Anyway, so I'm suggesting this:

```suggestion
  const errorColor = useMemo(
    () => defaults.variant === 'error' ? 'danger.500' : 'primary.500', 
    [defaults.variant]
  )
```

And then later in the markup, replace `errorIcon` by:
```javascript
<Icon as={BxsErrorCircle} fontSize="1rem" color={errorColor} />
```

and `errorText` by:
```javascript
<Text textColor={errorColor} noOfLines={1}>
  {defaults.message}
</Text>
```

Reply:
I see your point and I agree - i'll refactor to get a bit more modularity here!